src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -604,6 +611,48 @@ public function formAction($id, Request $request)
         ];
     }
 
+    /**
+     * Duplicate product info from one shop to another
+     *
+     * @param Product $product
+     * @throws \LogicException
+     */
+    private function duplicateProductFromDefaultShop(Product $product)
+    {
+        $productAdapter = $this->get('prestashop.adapter.data_provider.product');
+        $product = $productAdapter->getProduct($product->id, null, null, $product->id_shop_default);
+        $this->duplicateProductCombinationsFromDefaultShop($product);
+
+        return $product;
+    }
+    /**
+     * Duplicate combinations from one shop to another
+     *
+     * @param Product $product
+     */
+    private function duplicateProductCombinationsFromDefaultShop(Product $product)
```

Comment:
obviously we can't keep that here: object models calls are not allowed in the bundle: how about extract these 2 privates functions in a specific class with the responsability of duplication of a product?

Reply:
agree, will do :)

Reply:
or even `DuplicateProductCommand` with its handler. :stuck_out_tongue: